linbox/algorithms/lifting-container.h

```diff
@@ -152,12 +152,15 @@ namespace LinBox
             Integer N, D, L, Prime;
             this->_intRing.convert(Prime,_p);
 
-            auto hadamardBound = RationalSolveHadamardBound(A, b);
-            N = Givaro::pow(Integer(2), hadamardBound.numBoundBitSize);
-            D = Givaro::pow(Integer(2), hadamardBound.denBoundBitSize);
-			L = N * D * 2;
-
-			_length = (size_t)logp(L,Prime) + 1;   // round up instead of down
+            auto hb = RationalSolveHadamardBound(A, b);
+            N = Givaro::pow(Integer(2), static_cast<size_t>(std::ceil(hb.numLogBound)));
```

Comment:
Maybe you can directly replace the pow by the 1<< here (and not in a future PR)

Reply:
Forgot that, thanks. Fixed